### ISSUE TITLE
Dropped Targets and tuner Gain update

### ISSUE THREAD

**Issue by D35Bonanza** [dndx/pitot#7]
Both es.rs and uat.rs in src/sensor/sdr helped reception, S/N is much better it appears. 480db to 402db
const TUNER_GAIN: i32 = 402;

gld90.rs in src/sensor/protocal
const MAX_STALE_SECS: u64 = 12; // do not report data more than 6 sec old
//dropping targets at 6sec, 12 sec I hang to target ~150miles on Es and uat no dropped targets and appear to pick up a tower in daylight rather then night which is much easier to do.

**Comment by dndx** [comment]
Hi,

6 seconds seems to be a little too short? For example, FIS-B targets relies on primary reday return and are usually generated between 7-10 seconds.

Having `MAX_STALE_SECS` too small may cause targets on some EFB (at least FF) to flap.

For the `TUNER_GAIN`, I suspect what I use is a bit too high but that is basically what Stratux uses as well. What's the improvement on reducing it to `40.2dB`?

**Comment by D35Bonanza** [comment]
Hello dndx
The slight adjustment in gain will give you a higher Signal/noise ratio. I got thinking about the SDR noise at full gain. Most Fltaware users set there SDRs to 40.2 db to start and it appears to be the best S/N when Plotted on SDSharp. On Avare, I get more targets then I get with Startux with the updated settings.
The MAX_STALE_SECS at 6sec was dropping or never getting targets for some reason.

Stratux have always used the highest gain setting however they also use so called high gain antenna for some reason.

My ground station is at ~500ft MSL.(Woodside) I now get targets with stock SDR antennas ~150nm both ES and UAT. The ~alum rock UAT tower is received when on.

I just started looking at the rust code. I now have my rpi-img-builder working great. I can compile in Debian Stretch Xeon or Rpi Jessie ARM. I needed to use  pkg-config to build correctly, isc-DCHP-client and resolvconf to get eth0 working correctly.
I have included xserver.xorg mate, lightdm vnc4server xrdp for my development setup On first startup you must issue as root ln -s /dev/resolvconf/run/resolv.conf /etc/resolv.conf the DNS server working.

I'll keep checking it out as time allows

**Comment by dndx** [comment]
@D35Bonanza Yeah. I agree setting gains to their maximum was never a good idea, but since I thought it worked well enough, never really tested with lower gains.

I will give `40.2 dB` a try next time I fly. As long as there is no substantial degrade of performance, I am perfectly happy with that.

So with the 6 sec `MAX_STALE_SECS`, Avare will drop targets? Seems that it does not display stale targets as long as FF (which is more than 10 sec).

Interestingly I am also able to receive the UAT station near OSI in my home, so we must live close.

**Comment by dndx** [comment]
@D35Bonanza Just checked, looks like PiAware is using AGC in their `dump1090`:

```
/usr/bin/dump1090-fa --device-index 0 --gain -10 ...
```

So you think it is worth trying in this use case?

**Comment by D35Bonanza** [comment]
Yes I know, this is why users in the field stopped using AGC and set the gain to 40db to check it out for S/N using sdsharp
AGC may work or may not work so well. The rpi does have noise on it own.  A pc on a desktop or laptop have far less EMI output.  Other have used a BiasT and powered LNA at antenna feed and tuned the gain way down for the best S/N
